src/main/java/pl/kulinarnie/demo/MainController.java

```diff
@@ -0,0 +1,160 @@
+package pl.kulinarnie.demo;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+
+@Controller
+public class MainController {
+
+    private CategoryRepository categoryRepository;
+    private RecipeRepository recipeRepository;
+
+    public MainController(CategoryRepository categoryRepository, RecipeRepository recipeRepository) {
+        this.categoryRepository = categoryRepository;
+        this.recipeRepository = recipeRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model) {
+
+        List<Category> categories = categoryRepository.findAll();
+        model.addAttribute("categories", categories);
+
+        return "home";
+    }
+
+
+    @GetMapping("/allrecipies")
+    public String allRecipies(Model model) {
+
+        List<Category> categories = categoryRepository.findAll();
+        model.addAttribute("categories", categories);
+        List<Recipe> recipeListAll = recipeRepository.findAll();
+        model.addAttribute("recipies", recipeListAll);
+
+        return "allRecipies";
+    }
+
+
+    @GetMapping("/find")
+    public String search(Model model, @RequestParam String keyword) {
+
+        List<Category> categories = categoryRepository.findAll();
+        model.addAttribute("categories", categories);
+
+        List<Recipe> recipeListAll = recipeRepository.findAllByKeyword(keyword);
+        if (recipeListAll.isEmpty()) {
+
+            String keywordFirstLetter = keyword.substring(0, 1);
+            String keywordFirstLetterCaps = keywordFirstLetter.toUpperCase();
+            String key = keywordFirstLetterCaps.concat(keyword.substring(1, keyword.length()));
+            recipeListAll = recipeRepository.findAllByKeyword(key);
+        }
+        if (keyword != null) {
+            model.addAttribute("recipies", recipeListAll);
+        }
+        return "allRecipies";
+    }
+
+
+    @GetMapping("/category")
+    public String getRecipies(Model model, @RequestParam(defaultValue = "5") Long id) {
```

Comment:
Nie powinieneś tutaj mieć raczej żadnej wartości domyślnej. Nie wiadomo co oznacza wartość 5. Jeżeli ktoś nie przekaże parametru id, to powinno być przekierowanie do strony błędu, albo głównej